elastic/kibana
# [ResponseOps][Cases] Allow users to create case using templates

## Summary

Fixes https://github.com/elastic/kibana/issues/181309

This PR 
- allows users to create, edit or delete templates via cases > settings page
- allows users to create case using templates

https://github.com/elastic/kibana/assets/117571355/39226aa4-9d9a-41a8-a900-ca765ed98e1b

## Testing

1. Go to all solutions and create cases with all fields (including all fields of all supported connectors) without using templates. Verify that everything is working as expected.
2. Go to all solutions and create and edit templates with various fields. Verify that everything is working as expected.
3. Go to all solutions, create different templates on each solution, and verify that when creating a case you can use templates and everything is working as expected.
4. Go to the alerts table of o11y and security and attach alerts to a new case. Verify that in the flyout the templates are working as expected.
5. Go to ML and try to attach an ML visualization to a new case. Verify that the solution picker is working as expected and it resets the form when changing solutions.
6. Create a template with custom fields. Delete one of the custom fields from the settings page. Verify that it is also deleted from the template.

### Checklist

Delete any items that are not applicable to this PR.


**Flaky test runner**: https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/6425

### For maintainers



## Release notes
Allow users to create case using templates.
